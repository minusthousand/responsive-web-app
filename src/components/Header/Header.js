import React from 'react'
import { useState } from 'react'
import useScrollEvent from '../../hooks/useScrollEvent'

import HeaderStyled from '../../layout/header/index'

function Header() {
    const [isMenuOpen, setIsMenuOpen] = useState(false)
    const [isHeaderVisible, setIsHeaderVisible] = useState(false)

    const toggleMobileMenu = () => setIsMenuOpen(!isMenuOpen)
    const closeMobileMenu = () => setIsMenuOpen(false)

    const scrollToHome = () => {
        window.scrollTo({
            top: 0,
            behavior: 'smooth',
        })
    }

    const handleNavigateHome = () => {
        closeMobileMenu()
        scrollToHome()
    }

    const changeNavbarVisibility = () => {
        if (window.scrollY >= 65) {
            return setIsHeaderVisible(true)
        }
        return setIsHeaderVisible(false)
    }

    useScrollEvent(changeNavbarVisibility)

    return (
        <HeaderStyled visibility={isHeaderVisible ? 'visible' : 'hidden'}>
            <HeaderStyled.Container>
                <HeaderStyled.Logo onClick={handleNavigateHome}>
                    PRODUCT PAGE
                    <HeaderStyled.Icon className="fas fa-store"></HeaderStyled.Icon>
                </HeaderStyled.Logo>
                <HeaderStyled.MenuIcon onClick={toggleMobileMenu}>
                    <HeaderStyled.Icon
                        menu
                        className={isMenuOpen ? 'fas fa-times' : 'fas fa-bars'}
                    />
                </HeaderStyled.MenuIcon>
                <HeaderStyled.Menu
                    isActive={isMenuOpen ? 'active' : 'unactive'}
                >
                    <HeaderStyled.Item onClick={handleNavigateHome}>
                        Home
                    </HeaderStyled.Item>
                </HeaderStyled.Menu>
            </HeaderStyled.Container>
        </HeaderStyled>
    )
}

export default Header
