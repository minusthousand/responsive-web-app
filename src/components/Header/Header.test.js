import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import Header from './Header'

Object.defineProperty(window, 'scrollTo', {
    configurable: true,
})
window.scrollTo = jest.fn()

describe('<Header />', () => {
    it('should render', () => {
        const { container } = render(<Header />)

        expect(container).toMatchSnapshot()
    })

    it('should scroll to top when header logo was clicked', () => {
        const { getByText } = render(<Header />)

        fireEvent.click(getByText('PRODUCT PAGE'))

        expect(window.scrollTo).toBeCalled()
    })

    it('should set menu icon to fa-times when mobile menu is open', () => {
        const { getByTestId } = render(<Header />)

        fireEvent.click(getByTestId('menu-icon-wrapper'))

        expect(getByTestId('menu-icon').className).toBe('fas fa-times')
    })

    it('should set menu icon to fa-bars when mobile menu is closed', () => {
        const { getByTestId } = render(<Header />)

        expect(getByTestId('menu-icon').className).toBe('fas fa-bars')
    })

    it('should scroll to top when home items was clicked', () => {
        const { getByText } = render(<Header />)

        fireEvent.click(getByText('Home'))

        expect(window.scrollTo).toBeCalled()
    })
})
