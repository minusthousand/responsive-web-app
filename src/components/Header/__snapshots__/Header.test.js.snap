// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render 1`] = `
<div>
  <nav
    class="hidden"
  >
    <div
      class="header-container"
    >
      <div
        class="header-logo"
      >
        PRODUCT PAGE
        <i
          class="fas fa-store"
        />
      </div>
      <div
        class="menu-icon"
        data-testid="menu-icon-wrapper"
      >
        <i
          class="fas fa-bars"
          data-testid="menu-icon"
        />
      </div>
      <ul
        class="nav-menu"
      >
        <li
          class="nav-item nav-links"
        >
          Home
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
